message.go

```diff
@@ -249,7 +250,7 @@ func (m *Message) wrappErrorUnpack(src []byte) error {
 // unpack contains the core logic for unpacking the message. This method does
 // not handle locking or error wrapping and should typically be used internally
 // after ensuring concurrency safety.
-func (m *Message) unpack(src []byte) error {
+func (m *Message) unpack(src []byte) (string, error) {
```

Comment:
Could we just create the `UnpackError` in `unpack`, and remove `wrappErrorUnpack`?
It doesn't look like `unpack` is used anywhere else in this file, and it would make the method signature clearer